mobile/examples/speech_recognition/ios/SpeechRecognition/AudioRecorder.swift

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import AVFoundation
+import Foundation
+
+private let kSampleRate: Int = 16000
+private let kRecordingDuration: TimeInterval = 10
+
+class AudioRecorder {
+  typealias RecordingBufferAndData = (buffer: AVAudioBuffer, data: Data)
+  typealias RecordResult = Result<RecordingBufferAndData, Error>
+  typealias RecordingDoneCallback = (RecordResult) -> Void
+
+  enum AudioRecorderError: Error {
+    case Error(message: String)
+  }
+
+  func record(callback: @escaping RecordingDoneCallback) {
+    let session = AVAudioSession.sharedInstance()
+    session.requestRecordPermission { allowed in
+      do {
+        guard allowed else {
+          throw AudioRecorderError.Error(message: "Recording permission denied.")
+        }
+
+        try session.setCategory(.record)
+        try session.setActive(true)
+
+        let tempDir = FileManager.default.temporaryDirectory
+
+        let recordingUrl = tempDir.appendingPathComponent("recording.wav")
+
+        let formatSettings: [String: Any] = [
+          AVFormatIDKey: kAudioFormatLinearPCM,
+          AVSampleRateKey: kSampleRate,
+          AVNumberOfChannelsKey: 1,
+          AVLinearPCMBitDepthKey: 16,
+          AVLinearPCMIsBigEndianKey: false,
+          AVLinearPCMIsFloatKey: false,
+          AVEncoderAudioQualityKey: AVAudioQuality.high.rawValue,
+        ]
+
+        let recorder = try AVAudioRecorder(url: recordingUrl, settings: formatSettings)
+        self.recorder = recorder
+
+        let delegate = RecorderDelegate(callback: callback)
+        recorder.delegate = delegate
+        self.recorderDelegate = delegate
+
+        guard recorder.record(forDuration: kRecordingDuration) else {
+          throw AudioRecorderError.Error(message: "Failed to record.")
+        }
+
+        // control should resume in recorder.delegate.audioRecorderDidFinishRecording()
+      } catch {
+        callback(.failure(error))
+      }
+    }
+  }
+
+  private var recorderDelegate: RecorderDelegate?
+  private var recorder: AVAudioRecorder?
+
+  private class RecorderDelegate: NSObject, AVAudioRecorderDelegate {
+    private let callback: RecordingDoneCallback
+
+    init(callback: @escaping RecordingDoneCallback) {
+      self.callback = callback
+    }
+
+    func audioRecorderDidFinishRecording(
+      _ recorder: AVAudioRecorder,
+      successfully flag: Bool
+    ) {
+      print("audioRecorderDidFinishRecording")
```

Comment:
Is this a message will be displayed on the screen? or just used for debugging the app or so..

Reply:
extra debugging info, will remove it